### PR TITLE
Update functions.py with select_columns function

### LINKED ISSUE #77: Select_Columns Function Added Suggestion...
I see there is remove columns function.  I think a select_columns function would work nice.  It would be cleaner and easier to understand then df[['col1',col2','col3']].

### PR DESCRIPTION
**Pull request by jcvall**
Added select_columns function with an option to invert columns that will alternatively drop selected columns.

This PR resolves #77 .

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from <your_username>:master, but rather from <your_username>:<branch_name>.
2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [x] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - pycodestyle checkin
    - running the test suite
    - docs build

## Code Changes

If you are adding code changes, please ensure the following:

- [x] Ensure that you have added tests.
- [x] Run all tests (`$ pytest .`) locally on your machine.
    - [x] Check to ensure that test coverage covers the lines of code that you have added.
    - [x] Ensure that all tests pass.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [ ] Build the docs locally.
- [ ] View the docs to check that it renders correctly.

# PR Description

Please describe the changes proposed in the pull request:

- 
- 
- 

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by ericmjl** [comment]
@jcvall thanks for the PR - it's a great start!! :smile: 

The automatic checks that are run on TravisCIare part of a suite of things that I get program Travis to run every time a new PR is made. This ensures that the code is properly tested, formatted according to a consistent style, and more. I also manually check the docstrings to ensure that they are written properly.

We'll go one-by-one, so things aren't too overwhelming for you.

The first thing you'll want to do is to write at least one test case for the function.

This checklist item exists to ensure that every new function that is implemented comes accompanied by a test case. This acts like a "contract" of sorts between the person who wrote the code, and future developers (including the code writer him/herself). It goes into an automatic test suite that you can run.

To write a test, you will want to go into the `tests/` directory. There, you will see a `test_functions.py` file. In there, if you read the code, you'll get a sense of the pattern of software testing that's used with pyjanitor. Basically, you write a function that tests one "atomic" aspect of the function, and nothing else. 

Please go ahead and add some tests! Once that's done, I'll loop back on code style and more.

**Comment by jcvall** [comment]
Not sure why these asserts are not working....feedback is welcome.

File: tests/test_functions.py [hunk 1]
```diff
@@ -872,3 +872,24 @@ def test_reset_index_inplace_drop(dataframe):
...
     )
+
+        
+def test_select_columns(dataframe):
+    df = dataframe.select_columns(columns= [
+        "a",
+        "Bell__Chart",
+        "cities"
+    ])
+    assert df is dataframe[[
```
Comment (ericmjl) [inline tests/test_functions.py]: I think I see what the issue might be! If you look above, there's a few pandas testing functions that we use.

Here's a suggestion, taken from the [pandas testing utilities](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.testing.assert_frame_equal.html):

```suggestion
    assert pd.testing.assert_frame_equal(df, dataframe[columns])
```

File: tests/test_functions.py [hunk 2]
```diff
@@ -872,3 +872,24 @@ def test_reset_index_inplace_drop(dataframe):
         dataframe.reset_index(drop=True),
         dataframe.reset_index_inplace(drop=True),
     )
+
+        
+def test_select_columns(dataframe):
+    df = dataframe.select_columns(columns= [
```
Comment (ericmjl) [inline tests/test_functions.py]: To make the below fix work, I would suggest doing:

```suggestion
    columns = ['a', 'Bell__Chart', 'cities']
    df = dataframe.select_columns(columns=columns)
```

File: tests/test_functions.py [hunk 3]
```diff
@@ -872,3 +872,24 @@ def test_reset_index_inplace_drop(dataframe):
...
+        "cities"
+    ]]
+    
+def test_select_columns_invert(dataframe):
+    df = dataframe.select_columns(columns= [
+        "a",
+        "Bell__Chart",
+        "cities"
+    ],invert=True)
+    assert df is dataframe[["decorated-elephant","animals@#$%^"]]
```
Comment (ericmjl) [inline tests/test_functions.py]: This would be fixable using the aforementioned pattern as well.

**Comment by jcvall** [comment]
iMac:pyjanitor jcvall$ make check
black -l 79 .
make: black: No such file or directory
make: *** [check] Error 1
John-Valliers-iMac:pyjanitor jcvall$ flake8 janitor test
janitor/functions.py:9:1: F401 'warnings.warn' imported but unused
janitor/functions.py:1666:4: W291 trailing whitespace
janitor/functions.py:1674:1: W293 blank line contains whitespace
janitor/functions.py:1679:80: E501 line too long (91 > 79 characters)
janitor/functions.py:1683:1: W293 blank line contains whitespace
janitor/functions.py:1685:1: W293 blank line contains whitespace
janitor/functions.py:1689:27: W292 no newline at end of file
janitor/.ipynb_checkpoints/functions-checkpoint.py:9:1: F401 'warnings.warn' imported but unused
test:1:1: E902 FileNotFoundError: [Errno 2] No such file or directory: 'test'
John-Valliers-iMac:pyjanitor jcvall$ py.test
Traceback (most recent call last):
  File "/Users/jcvall/anaconda/bin/py.test", line 7, in <module>
    from pytest import main
  File "/Users/jcvall/anaconda/bin/pytest.py", line 14, in <module>
    from testing import *
ModuleNotFoundError: No module named 'testing'

--- So I made the whitespace changes but the 'make check' doesn't seem to run.  Thanks for your patients here, I feel like I am learning a lot though.  Could I be doing something wrong above?

**Comment by ericmjl** [comment]
@jcvall no worries about the error, I'm glad the experience has been educational so far!

Let me see, have you created a virtual environment (venv) or conda environment for development? My current working hypothesis is that these tools aren't found in your current path.

If my hypothesis is correct, then you would have to create a virtual environment that contains black, pytest, pycodestyle, flake8, etc. Have you done this before? If you're using conda, I have provided an environment file that you can use, by entering the following commands at the terminal:

```bash
$ conda env create -f environment.yml  # create environment based on environment.yml
$ conda activate pyjanitor-dev  # activate the environment
```

Let me know if you're already inside a conda environment for development. If so, we can continue debugging here.

You're doing great thus far! :smile: Just a bit more to go.

**Comment by jcvall** [comment]
Forgive me....So I have been giving it an honest try over a couple of days.  It took me a bit to figure out in the Mac env 'source activate pyjanitor-dev' was what I needed after conda env create -f environment-dev.yml, but I got there.  I have tried 'make check' with no luck even after setting up the env, see (1).  Also I ran the py.test with the changes suggested and got the follow errors, see (2).  I'm hanging in there and taking notes for next time I have a PR.  Suggestions on how I can move forward with these issues?

(1)
(pyjanitor-dev) John-Valliers-iMac:pyjanitor jcvall$ make check
black -l 79 .
make: black: No such file or directory
make: *** [check] Error 1

(2)
=================================== FAILURES ===================================
_____________________________ test_select_columns ______________________________

dataframe =    a  Bell__Chart  decorated-elephant animals@#$%^     cities
0  1     1.234523                   1       rabbit  Camb...     2.456234                   2      leopard   Shanghai
8  3     3.234612                   3         lion      Basel

    def test_select_columns(dataframe):
        columns = ['a', 'Bell__Chart', 'cities']
        df = dataframe.select_columns(columns=columns)
    
>       assert pd.testing.assert_frame_equal(df, dataframe[columns])
E       AssertionError: assert None
E        +  where None = <function assert_frame_equal at 0x1108fd9d8>(   a  Bell__Chart     cities\n0  1     1.234523  Cambridge\n1  2     2.456234   Shanghai\n2  3     3.234612      Basel\n3 ...hai\n5  3     3.234612      Basel\n6  1     1.234523  Cambridge\n7  2     2.456234   Shanghai\n8  3     3.234612      Basel,    a  Bell__Chart     cities\n0  1     1.234523  Cambridge\n1  2     2.456234   Shanghai\n2  3     3.234612      Basel\n3 ...hai\n5  3     3.234612      Basel\n6  1     1.234523  Cambridge\n7  2     2.456234   Shanghai\n8  3     3.234612      Basel)
E        +    where <function assert_frame_equal at 0x1108fd9d8> = <module 'pandas.testing' from '/Users/jcvall/anaconda/envs/pyjanitor-dev/lib/python3.6/site-packages/pandas/testing.py'>.assert_frame_equal
E        +      where <module 'pandas.testing' from '/Users/jcvall/anaconda/envs/pyjanitor-dev/lib/python3.6/site-packages/pandas/testing.py'> = pd.testing

tests/test_functions.py:881: AssertionError
__________________________ test_select_columns_invert __________________________

dataframe =    a  Bell__Chart  decorated-elephant animals@#$%^     cities
0  1     1.234523                   1       rabbit  Camb...     2.456234                   2      leopard   Shanghai
8  3     3.234612                   3         lion      Basel

    def test_select_columns_invert(dataframe):
        columns = ['a', 'Bell__Chart', 'cities']
        df = dataframe.select_columns(columns=columns,invert=True)
        columns2 = ["decorated-elephant","animals@#$%^"]
>       assert pd.testing.assert_frame_equal(df, dataframe[columns2])
E       AssertionError: assert None
E        +  where None = <function assert_frame_equal at 0x1108fd9d8>(   decorated-elephant animals@#$%^\n0                   1       rabbit\n1                   2      leopard\n2            ...3         lion\n6                   1       rabbit\n7                   2      leopard\n8                   3         lion,    decorated-elephant animals@#$%^\n0                   1       rabbit\n1                   2      leopard\n2            ...3         lion\n6                   1       rabbit\n7                   2      leopard\n8                   3         lion)
E        +    where <function assert_frame_equal at 0x1108fd9d8> = <module 'pandas.testing' from '/Users/jcvall/anaconda/envs/pyjanitor-dev/lib/python3.6/site-packages/pandas/testing.py'>.assert_frame_equal
E        +      where <module 'pandas.testing' from '/Users/jcvall/anaconda/envs/pyjanitor-dev/lib/python3.6/site-packages/pandas/testing.py'> = pd.testing

tests/test_functions.py:888: AssertionError
===================== 2 failed, 93 passed in 3.38 seconds ======================

**Comment by ericmjl** [comment]
@jcvall, thanks for pinging back! You've broken down the problem well, and yes, the pasted commands are very helpful for debugging!

For issue (1), there are a few more commands that you can try, for debugging purposes.

```
$ which black
```

This queries your [PATH environment variable](https://en.wikipedia.org/wiki/PATH_(variable)) for the `black` program. In general, I think it should be kept inside your project environment (i.e. `pyjanitor-dev`) here. If you could paste the output of the aforementioned command, we can figure out whether or not `black` was correctly installed. 

If `which` wasn't able to find a `black` binary, there's a way to fix this. Firstly, activate the pyjanitor-dev environment:

```
$ source activate pyjanitor-dev
```

Then, install black:

```
$ conda install -c conda-forge black
```

Finally, verify that black has been installed:

```
$ which black
```

On the second issue, I may have misread the code for pandas testing. As I re-read [the docs](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.testing.assert_frame_equal.html), I realized it doesn't return anything -- instead, it contains the assert statement within itself! My bad.

Hence, we can get rid of the `assert` statement. The code should look like:

```
# incorrect:
# assert pd.assert_frame_equal(........)

# correct:
pd.assert_frame_equal(........)
```

Let's see what happens when that is changed! :smile:

**Comment by jcvall** [comment]
This is a beautiful thing.   :)  I owe you a lot of thanks for getting me this far. It seem to work.

(pyjanitor-dev) John-Valliers-iMac:pyjanitor jcvall$ make check
black -l 79 .
reformatted /Users/jcvall/Documents/GitHub/pyjanitor/janitor/.ipynb_checkpoints/functions-checkpoint.py
reformatted /Users/jcvall/Documents/GitHub/pyjanitor/janitor/functions.py
reformatted /Users/jcvall/Documents/GitHub/pyjanitor/tests/test_functions.py
All done! ✨ 🍰 ✨
3 files reformatted, 11 files left unchanged.
pytest
============================= test session starts ==============================
platform darwin -- Python 3.6.6, pytest-4.0.2, py-1.7.0, pluggy-0.8.0
rootdir: /Users/jcvall/Documents/GitHub/pyjanitor, inifile: pytest.ini
plugins: cov-2.6.0
collected 95 items                                                             

tests/test_biology.py .                                                  [  1%]
tests/test_df_registration.py .................                          [ 18%]
tests/test_functions.py ................................................ [ 69%]
.............................                                            [100%]

========================== 95 passed in 3.59 seconds ===========================
pycodestyle .
cd docs && make html
Running Sphinx v1.8.2
making output directory...
loading intersphinx inventory from https://docs.python.org/objects.inv...
intersphinx inventory has moved: https://docs.python.org/objects.inv -> https://docs.python.org/3/objects.inv
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 5 source files that are out of date
updating environment: 5 added, 0 changed, 0 removed
reading sources... [100%] index                                                 
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index                                                  
generating indices... genindex py-modindex
highlighting module code... [100%] janitor.functions                            
writing additional pages... search
copying static files... WARNING: html_static_path entry '/Users/jcvall/Documents/GitHub/pyjanitor/docs/_static' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 1 warning.

**Comment by ericmjl** [comment]
@jcvall, thanks for the contribution, everything looks great to me at the moment! :smile: Going to merge in and cut a new release.

**Pull request merged**